command/genesis/polybft_params.go

```diff
@@ -645,6 +696,74 @@ func (p *genesisParams) validateBladeAdminFlag() error {
 	return nil
 }
 
+// validatePremineInfo validates whether reserve account (0x0 address) is premined
+func (p *genesisParams) validatePremineInfo() error {
+	isZeroAddressPremined := false
+
+	for _, premineInfo := range p.premineInfos {
+		if premineInfo.Address == types.ZeroAddress {
+			isZeroAddressPremined = true
+		} else if !p.nativeTokenConfig.IsMintable {
+			return errNoPremineAllowed
+		}
+	}
+
+	if !isZeroAddressPremined {
+		return errReserveAccMustBePremined
+	}
+
+	return nil
+}
+
+// validateBurnContract validates burn contract. If native token is mintable,
+// burn contract flag must not be set. If native token is non mintable only one burn contract
+// can be set and the specified address will be used to predeploy default EIP1559 burn contract.
+func (p *genesisParams) validateBurnContract() error {
+	if p.isBurnContractEnabled() {
+		burnContractInfo, err := parseBurnContractInfo(p.burnContract)
+		if err != nil {
+			return fmt.Errorf("invalid burn contract info provided: %w", err)
+		}
+
+		if p.nativeTokenConfig.IsMintable {
+			if burnContractInfo.Address != types.ZeroAddress {
+				return errors.New("only zero address is allowed as burn destination for mintable native token")
+			}
+		} else {
+			if burnContractInfo.Address == types.ZeroAddress {
+				return errors.New("it is not allowed to deploy burn contract to 0x0 address")
+			}
+		}
+	}
+
+	return nil
+}
+
+// validateStakeInfo validates and parses stake flag
+func (p *genesisParams) validateStakeInfo() error {
+	if !p.nativeTokenConfig.IsMintable && len(p.stake) > 0 {
+		return errNoStakeAllowed
+	}
```

Comment:
What should we do with `stakeInfo` in case stake token is non-native token? 

Should we in that use case `mint` stake token to those accounts in the `contracts_initializer.go`?

Also, this condition should be relaxed a bit, `if stakeToken != nativeToken`, we still should allow stake info right?
```suggestion
	if p.stakeTokenAddr == contracts.NativeERC20TokenContract && !p.nativeTokenConfig.IsMintable && len(p.stake) > 0 {
		return errNoStakeAllowed
	}
```

Reply:
Let's address this through a separate PR.